users/models.py

```diff
@@ -1,15 +1,34 @@
+from urllib.parse import urlencode
+
+from django.conf import settings
 from django.contrib.auth.models import AbstractUser
 from django.core.exceptions import ValidationError
 from django.db import models, IntegrityError
 from django.db.models import Q
+from django.template.loader import render_to_string
+from django.core.mail import send_mail
 from slack.errors import SlackApiError
 
 from common.utils import pprint_obj, get_slack_client
 
 
 class User(AbstractUser):
-    # Eventually, we will add fields here to extend this class
-    pass
+    is_verified = models.BooleanField(default=False)
+
+    def send_verification_email(self, token):
+        context = {
+            'verification_url': build_email_verification_url(self.email, token),
+            'first_name': self.first_name
+        }
+        text_email = render_to_string('users/verify_email.txt', context)
+        html_email = render_to_string('users/verify_email.html', context)
```

Comment:
I wonder if we shouldn't put all of this in sendgrid and use template vars to replace dynamic data like the link or names